app/controllers/works_controller.rb

```diff
@@ -785,6 +785,17 @@ def update_multiple
     end
   end
 
+  # Reindex the work.
+  def reindex
+    if logged_in_as_admin? || permit?("tag_wrangler")
+      RedisSearchIndexQueue.queue_works([params[:id]], priority: :high)
+      flash[:notice] = ts('Work queued to be reindexed')
+    else
+      flash[:notice] = ts("Sorry, you don't have permission to access the page you were trying to reach.")
```

Comment:
I'd go with something like "to perform this action" or "to reindex works" or something, since we're not really trying to access a page